.github/workflows/integration-test.yml

```diff
@@ -247,6 +247,7 @@ jobs:
           # re-install zenml to be sure
           poetry install --extras server
           # install airflow requirements
+          pip install numpy==1.23.5
```

Comment:
Could you explain why this is needed?

Reply:
Ah, so this is an issue now because we use newer MLflow versions, which, in turn, use newer NumPy? In that case, how about we add `numpy<1.24.0` as an integration requirement explicitly into the MLflow integration?

The change above made seem as if this was related to Airflow somehow, that's why I was confused

Reply:
I tried to fix the NumPy version on the MLFlow requirements but it didn't work! exactly as you said the upgrading of `numpy`happnes at stage 3 of the test which is related to `airflow` but it only start happening after a larger range of `mlflow`versions was supported! Should we leave it in the integration-test.yml for now? @fa9r